src/idm/entity.js

```diff
@@ -0,0 +1,175 @@
+    * @fulfil {Array} all entities with a given type
+    * @returns {Promise}
+    * @example
+    * agile.idm.entity.getEntitiesByAttributeValue([{attribute_typeattribute_type:"credentials.dropbox","attribute_value":"expected attribute value for dropbox credentials"}]).then(function(entities) {
+    *   console.log(entities);
+    * });
+    **/
+    getEntitiesByAttributeValue: (constraints) => {
+      let url = `${base}/api/v1/entity/search`;
+      return axios({
+        method: 'POST',
+        url: url,
+        headers: {"Authorization":`bearer ${token}`},
+        data:{ "criteria":constraints}
+      })
+      .then(res => (res.data))
+      .catch(errorHandler);
+    },
+    /**
+    * @summary get Entity by entity id and type
+    * @name getEntity
+    * @public
+    * @function
+    * @memberof agile.idm.entity
+    * @param {String} entityId - id of entity
+    * @param {String} entityType - type of entity
+    * @fulfil {Object} entity entity
+    * @returns {Promise}
+    * @example
+    * agile.idm.entity.getEntity('1','/sensor').then(function(result) {
+    *   console.log('entity created!'+result);
+    * });
+    **/
+    getEntity: (entity_id, entity_type) => axios({
+      method: 'get',
+      url: `${base}/api/v1/entity/${entity_type}/${entity_id}`,
+      headers: {"Authorization" : `bearer ${token}`},
+
+    })
+    .then(res => (res.data))
+    .catch(errorHandler),    /**
+        * @summary Create a group onwned by the authenticated user
+        * @name createEntity
+        * @public
+        * @function
+        * @memberof agile.idm.entity
+        * @param {String} entityId - id of entity
+        * @param {String} entityType - type of entity
+        * @param {object} entity - An object containing the entity
+        * @fulfil {Object} entity created
+        * @returns {Promise}
+        * @example
+        * agile.idm.entity.createEntity('1','/sensor',{"name":"entity's name"}).then(function(result) {
+        *   console.log('entity created!'+result);
+        * });
+        **/
+        createEntity: (entity_id, entity_type, entity) => axios({
+          method: 'POST',
+          url: `${base}/api/v1/entity/${entity_type}/${entity_id}`,
+          headers: {"Authorization" : `bearer ${token}`},
+          data: entity
+        })
+        .then(res => (res.data))
+        .catch(errorHandler),
+    /**
+    * @summary Delete entity
+    * @name deleteEntity
+    * @public
+    * @function
+    * @memberof agile.idm.entity
+    * @param {String} entityId - id of entity
+    * @param {String} entityType - type of entity
+    * @fulfil {Undefined}
+    * @returns {Promise}
+    * @example
+    * agile.idm.entity.deleteEntity('1','/sensor').then(function() {
+    *   console.log('group removed!');
+    * });
+    **/
+    deleteEntity: (entity_id, entity_type) => axios({
```

Comment:
This is the naming issue again. I think it should be `agile.entity.delete()` currently it's `agile.entity.deleteEntity()`.